File: lambda/policy_handler.js
exports.handler = async (event) => {
    console.log("Event:", JSON.stringify(event, null, 2));

    const actionGroup = event?.actionGroup;
    const apiPath = event?.apiPath;
    const parameters = event?.parameters || [];

    // Extract policyNumber from action group parameters
    const policyNumberParam = parameters.find(p => p.name === "policyNumber");
    const policyNumber = policyNumberParam?.value;

    if (!policyNumber) {
        return actionGroupResponse(actionGroup, apiPath, 400, {
            message: "Policy number not provided"
        });
    }

    const result = processPolicy(policyNumber);

    return actionGroupResponse(actionGroup, apiPath, 200, result);
};

function processPolicy(policyNumber) {
    // TODO: Replace with actual DB lookup or business logic
    return {
        policyNumber,
        policyStatus: "Active",
        holder: "<customer_name>"
    };
}

function actionGroupResponse(actionGroup, apiPath, statusCode, body) {
    return {
        actionGroup,
        apiPath,
        httpStatusCode: statusCode,
        responseBody: {
            "application/json": {
                body: JSON.stringify(body)
            }
        }
    };
}

File: lambda/policy_handler.py
import json


def lambda_handler(event, context):
    policy_number = event.get("Details", {}).get("Parameters", {}).get("policyNumber")

    if not policy_number:
        return {"status": "error", "message": "Policy number not provided"}

    # TODO: Add your policy lookup / processing logic here
    result = process_policy(policy_number)

    return {"status": "success", "policyNumber": policy_number, "result": result}


def process_policy(policy_number: str) -> dict:
    # Placeholder: replace with actual DB lookup or business logic
    return {"policyStatus": "Active", "holder": "<customer_name>", "policyNumber": policy_number}
